### PR TITLE
Re-add help_on_error for download-ci-llvm

### LINKED ISSUE #97503: Add back `help_on_error` for download-ci-llvm
I accidentally missed this when porting it to rustbuild: https://github.com/rust-lang/rust/pull/95170/files#diff-9e25a089f077eae8c8cefe9586cc07498e64cd950be4c954cb90e4245e5f9fc3L634-L651

It would be nice to add it back.

Mentoring instructions: Add a new `help_on_error` parameter to `Builder::download_component_helper` and pass it down from `download_ci_llvm`:
https://github.com/rust-lang/rust/blob/00bb4df1fbfdcc9249f8fd5e009b0e18d30c96bf/src/bootstrap/builder.rs#L872 https://github.com/rust-lang/rust/blob/00bb4df1fbfdcc9249f8fd5e009b0e18d30c96bf/src/bootstrap/native.rs#L182

This code moved around in https://github.com/rust-lang/rust/pull/96687 so you may want to base your changes off that PR until it lands.

@rustbot label +E-easy +E-mentor +A-rustbuild

### PR DESCRIPTION
**Pull request by spiral-ladder**
Closes #97503 

- Re-added `help_on_error` for `download_component()` and the downstream functions
- Removed dead code in `bootstrap.py`

Thanks @jyn514 for the helpful tips!

(first contribution here, please let me know if I missed anything out!)

### CODE REVIEW

File: src/bootstrap/bootstrap.py [hunk 1]
```diff
@@ -197,8 +197,6 @@ def run(args, verbose=False, exception=False, is_bootstrap=False, help_on_error=
     code = ret.wait()
     if code != 0:
         err = "failed to run: " + ' '.join(args)
-        if help_on_error is not None:
-            err += "\n" + help_on_error
         if verbose or exception:
```
Comment (jyn514) [inline src/bootstrap/bootstrap.py:200]: Please also remove this from the function parameters (for run, and for all functions which call it, recursively)

File: src/bootstrap/native.rs [hunk 2]
```diff
@@ -179,7 +179,22 @@ fn download_ci_llvm(builder: &Builder<'_>, llvm_sha: &str) {
     let filename = format!("rust-dev-nightly-{}.tar.xz", builder.build.build.triple);
     let tarball = rustc_cache.join(&filename);
     if !tarball.exists() {
-        builder.download_component(base, &format!("{}/{}", url, filename), &tarball);
+        if !tarball.exists() {
```
Comment (jyn514) [inline src/bootstrap/native.rs:182]: This check looks duplicated - no need to check it twice.

**Review by jyn514 — COMMENTED** [review]
Thanks for tackling this! Can you please squash the commits once you've addressed the review comments? There are instructions at https://rustc-dev-guide.rust-lang.org/git.html#advanced-rebasing.

File: src/bootstrap/native.rs (hunk 2 above)
Comment (binggh) [reply src/bootstrap/native.rs:182]: oops - that was an accident, thanks for the catch!

**Review by jyn514 — APPROVED** [review]
Looks great once you squash the commits, thanks!

**Pull request merged**